crates/bevy_math/src/primitives/dim2.rs

```diff
@@ -125,6 +125,94 @@ impl Ellipse {
     }
 }
 
+/// A primitive shape formed by the region between two circles, also known as a ring.
+#[derive(Clone, Copy, Debug, PartialEq)]
+#[cfg_attr(feature = "serialize", derive(serde::Serialize, serde::Deserialize))]
+#[doc(alias = "Ring")]
+pub struct Annulus {
+    /// The inner radius of the annulus
+    pub inner_radius: f32,
+    /// The outer radius of the annulus
+    pub outer_radius: f32,
```

Comment:
Since an annulus is defined by two concentric circles, I think it'd be useful to make these `Circles`:

```suggestion
    /// The inner circle of the annulus.
    pub inner_circle: Circle,
    /// The outer circle of the annulus.
    pub outer_circle: Circle,
```

(`_circle` can maybe be omitted?)

This way, we also automatically get more helpers like computing the diameter, perimeter, or area of the inner/outer circle instead of needing to add them for `Annulus` directly.

Reply:
Agreed!

Reply:
Refactor landed; there might be a better implementation of `closest_point` available now, but I haven't looked into it yet.

Reply:
Yeah I wouldn't worry about that. Well within "numerical instability" levels.